lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -5763,6 +5764,36 @@ void populateTritonToLLVMPatterns(mlir::LLVMTypeConverter &typeConverter,
 
 class ConvertTritonGPUToLLVM
     : public ConvertTritonGPUToLLVMBase<ConvertTritonGPUToLLVM> {
+
+private:
+  void decomposeBlockedToDotOperand(ModuleOp mod) {
```

Comment:
Seems like this is a temporary solution, but we will ultimately need the native `blocked -> dot_op` layout conversion support?

IIUC, I think this code currently won't be invoked for most gemm cases because the pipeline pass generates `block->shared` and `shared->dot` itself.

Reply:
Not a very temporary solution; I can't really think of a better way to do  `blocked -> dot_op`. We could maybe use less shared memory, but that would require more barriers -- which I believe we can assume to be strictly worse. Going through shared memory is a nice default fallback for unsupported conversions.  Eventually though, we will hopefully do more progressive lowering from TritonGPU to LLVM, and the way we handle conversions will change a lot.


> IIUC, I think this code currently won't be invoked for most gemm cases because the pipeline pass generates block->shared and shared->dot itself.

Yep